### PR TITLE
Add graphite output reporter

### PR DESCRIPTION
**Pull request by wankunde**
Add a graphite output reporter, so we can sink metrics into graphite.

### CODE REVIEW

File: src/main/java/com/uber/profiling/reporters/GraphiteOutputReporter.java [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+            int num = 1;
+            for (Map.Entry<String, Object> entry2 : value2.entrySet()) {
+              formattedMetrics
+                  .put(key + "." + key1 + "." + entry2.getKey() + "." + num, entry2.getValue());
+            }
+            num++;
+          }
+        }
+      } else {
+        formattedMetrics.put(key, value);
```
Comment (boy-uber) [inline src/main/java/com/uber/profiling/reporters/GraphiteOutputReporter.java]: This code path will be hit when value == null? Suggest checking value == null before checking it type.

File: src/main/java/com/uber/profiling/reporters/GraphiteOutputReporter.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        List listValue = (List) value;
+        if (!listValue.isEmpty() && listValue.get(0) instanceof String) {
+          List<String> metricList = (List<String>) listValue;
+          formattedMetrics.put(key, String.join(",", metricList));
+        } else if (!listValue.isEmpty() && listValue.get(0) instanceof Map) {
+          List<Map<String, Object>> metricList = (List<Map<String, Object>>) listValue;
+          int num = 1;
+          for (Map<String, Object> metricMap : metricList) {
+            String name = null;
+            if (metricMap.containsKey("name") && metricMap.get("name") != null && metricMap
```
Comment (boy-uber) [inline src/main/java/com/uber/profiling/reporters/GraphiteOutputReporter.java]: Feel you could call getFormattedMetrics recursively here?

File: src/main/java/com/uber/profiling/reporters/GraphiteOutputReporter.java (hunk 1 above)
Comment (wankunde) [reply src/main/java/com/uber/profiling/reporters/GraphiteOutputReporter.java]: Hi，@buy-uber , Thanks for your review.

I just refactored the code, and add a test case.
Can you review my code again?

Many Thanks.

Comment (boy-uber) [reply src/main/java/com/uber/profiling/reporters/GraphiteOutputReporter.java]: Cool, thanks for contributing here and making the refactoring! It looks good to me now. Let me merge your PR.

**Pull request merged**